src/net_processing.cpp

```diff
@@ -1661,6 +1677,11 @@ void static ProcessGetBlockData(CNode& pfrom, const CChainParams& chainparams, c
                 } else {
                     connman.PushMessage(&pfrom, msgMaker.Make(nSendFlags, NetMsgType::BLOCK, *pblock));
                 }
+            } else if (inv.IsMsgMWEBHeader()) {
+                if (pblock->GetHogEx() != nullptr && !pblock->mweb_block.IsNull()) {
```

Comment:
If this fails is the peer left hanging?

Reply:
Like the similar comments above, I'm really not 100% sure whether we should be disconnecting on failures. This is one of those cases where it should never happen unless the requesting node is improperly coded (requesting MWEB header when they have enough knowledge to know it _doesn't have_ an MWEB). So either way, we're probably fine. I just erred on the side of less complexity.

It might be worth revisiting this discussion once we have at least one light client that supports these messages. We can discuss with their dev teams to see what's most useful for them at that time.